### PR TITLE
 zsdcc: init at unstable-2018-02-24 

### PR DESCRIPTION
**Pull request by bignaux**
###### Motivation for this change

This provides only zsdcc   and  zsdcpp for z88dk. Not conflict with sdcc.
@bjornfor : can you have a look please since you are sdcc derivation maintainer ? 

###### Things done
tested with z88dk example.

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/compilers/sdcc/zsdcc.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{ stdenv, sdcc, fetchFromGitHub, ... }:
```
Comment (bjornfor) [inline pkgs/development/compilers/sdcc/zsdcc.nix]: Please remove ```...``` and list each needed input explicitly.

File: pkgs/development/compilers/sdcc/zsdcc.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# https://www.z88dk.org/wiki/doku.php?id=temp:front#sdcc1
+# https://github.com/z88dk/z88dk/blob/master/src/zsdcc/readme.md
+
+sdcc.overrideAttrs (oldAttrs: rec {
+  version = "2018-02-24";
+  name = "zsdcc-unstable-${version}"; # ${oldAttrs.version}
```
Comment (bjornfor) [inline pkgs/development/compilers/sdcc/zsdcc.nix]: I'm not happy about packaging unstable versions. To me, that means it's not ready for Nixpkgs / end users.

File: pkgs/development/compilers/sdcc/zsdcc.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  meta = oldAttrs.meta // {
```
Comment (bjornfor) [inline pkgs/development/compilers/sdcc/zsdcc.nix]: The "meta" section is usually at the end of the file.

**Review by bjornfor — CHANGES_REQUESTED** [review]
See inline comments.

**Comment by bjornfor** [comment]
Oh, and please squash the commits. (It looks like only one logical change to me.)

File: pkgs/development/compilers/sdcc/zsdcc.nix (hunk 2 above)
Comment (bignaux) [reply pkgs/development/compilers/sdcc/zsdcc.nix]: there is no release of zsdcc, and it's not target end users, but users of z88dk that wants to work with new clibs.

File: pkgs/development/compilers/sdcc/zsdcc.nix (hunk 1 above)
Comment (bignaux) [reply pkgs/development/compilers/sdcc/zsdcc.nix]: will i not lost the interest of derivate sdcc doing that ?

File: pkgs/development/compilers/sdcc/zsdcc.nix (hunk 2 above)
Comment (bignaux) [reply pkgs/development/compilers/sdcc/zsdcc.nix]: there is no release of zsdcc, and it's not target end users, but users of z88dk that wants to work with new clibs.

**Comment by aanderse** [comment]
@bignaux are you able to address the comments above?

File: pkgs/development/compilers/sdcc/zsdcc.nix (hunk 1 above)
Comment (bjornfor) [reply pkgs/development/compilers/sdcc/zsdcc.nix]: I don't understand that question.

**Comment by wamserma** [comment]
This PR seems abandoned. @bignaux would you mind closing?

**Comment by aanderse** [comment]
Easy enough to reopen PR if there is interest again.

**Pull request closed without merging**